### PR TITLE
Add support for gre tunnel connections on a transit gateway

### PR DESCRIPTION
**Pull request by sarah-ulmer**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccIBMTransitGatewayDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccIBMTransitGatewayDataSource_basic -timeout 700m
?   	github.com/IBM-Cloud/terraform-provider-ibm	[no test files]
=== RUN   TestAccIBMTransitGatewayDataSource_basic
--- PASS: TestAccIBMTransitGatewayDataSource_basic (323.87s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm	326.411s
testing: warning: no tests to run
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/internal/hashcode	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/internal/mutexkv	(cached) [no tests to run]
?   	github.com/IBM-Cloud/terraform-provider-ibm/version	[no test files]



$ make testacc TESTARGS='-run=TestAccIBMTransitGatewayConnection_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccIBMTransitGatewayDataSource_basic -timeout 700m
?   	github.com/IBM-Cloud/terraform-provider-ibm	[no test files]
=== RUN   TestAccIBMTransitGatewayConnection_basic
2021/06/22 14:05:47 Inside testAccCheckIBMTransitGatewayConnectionExists :  
--- PASS: TestAccIBMTransitGatewayConnection_basic (544.68s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm	547.029s
testing: warning: no tests to run
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/internal/hashcode	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/internal/mutexkv	(cached) [no tests to run]
?   	github.com/IBM-Cloud/terraform-provider-ibm/version	[no test files]
```

### CODE REVIEW

**Review by kennburger — APPROVED** [review]

**Comment by hkantare** [comment]
Thanks for rasining PR...
Can you please check all are failed

**Comment by sarah-ulmer** [comment]
@hkantare could you please review again?

The checks failed because it needed to wait for a networking-go-sdk release

**Comment by hkantare** [comment]
Thanks ...for PR
1)Can you rebase with latest master code 
2) Please add all new arguments/attributes in the respecrtive doc file website/docs/d for datasources website/docs/r for resources
3)Please attach the screenhsot of testcases execution results

File: website/docs/d/tg_gateway.html.markdown [hunk 1]
```diff
@@ -45,15 +45,24 @@ In addition to all arguments above, the following attributes are exported:
   * `id` - The unique identifier of this gateway.
   * `status` - Gateway status.
   * `resource_group` - Resource group identifier.
-  * `connections` 
+  * `connections`  - All connections in the gateway.
     * `name` - The user-defined name for this transit gateway connection.
-    * `network_type` -  Defines what type of network is connected via this connection.Possible values: [classic,vpc]. 
+    * `network_type` -  Defines what type of network is connected via this connection.Possible values: [classic,vpc,gre+tunnel].
```
Comment (kennburger) [inline website/docs/d/tg_gateway.html.markdown:50]: Should be gre_tunnel ?

Comment (sarah-ulmer) [reply website/docs/d/tg_gateway.html.markdown]: Yes, and I fixed the typo

**Comment by sarah-ulmer** [comment]
@hkantare I rebased, updated the docs, and attached acceptance test results at the top of the issue. Could you please take another look and merge it?

**Pull request merged**